Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -200,36 +220,56 @@ public struct AbstractInterpreter {
         }
         return result
     }
-    
-    private mutating func executeEffects(_ instr: Instruction) {
+
+    private mutating func executeEffects(_ instr: Instruction) -> [(Variable, Type)] {
+        var typeChanges = [(Variable, Type)]()
+
         switch instr.op {
             
         case let op as LoadBuiltin:
-            set(instr.output, environment.type(ofBuiltin: op.builtinName))
+            typeChanges.append(
```

Comment:
This feels a bit bloated, maybe define a local helper function to reduce all of these back to a single function call/a single line:
```
func set(_ v: Variable, to type; Type) {
    setType(of: v, to: Type)
    typeChanges.append((v, t))
}
```
Then you can also remove the somewhat unintuitive return value of `setType`